### PR TITLE
mcux: add PIT driver

### PR DESCRIPTION
**Pull request by hakehuang**
add PIT driver support in Kenetis selected by CONFIG_COUNTER_MCUX_PIT
this will be use as hardware trigger source for dma

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

### CODE REVIEW

**Comment by hakehuang** [comment]
@MaureenHelm PIT driver will be used in PIT to ADC hardware trigger source to create a stable sampling rate

File: mcux/drivers/kinetis/CMakeLists.txt [hunk 1]
```diff
@@ -36,6 +36,7 @@ zephyr_library_sources_ifdef(CONFIG_SPI_MCUX_LPSPI	fsl_lpspi.c)
 zephyr_library_sources_ifdef(CONFIG_UART_MCUX		fsl_uart.c)
 zephyr_library_sources_ifdef(CONFIG_UART_MCUX_LPSCI	fsl_lpsci.c)
 zephyr_library_sources_ifdef(CONFIG_UART_MCUX_LPUART	fsl_lpuart.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER_MCUX_PIT	fsl_pit.c)
```
Comment (MaureenHelm) [inline mcux/drivers/kinetis/CMakeLists.txt:39]: Please move this up to line 18 to preserve alpha order by CONFIG symbol.

Comment (hakehuang) [reply mcux/drivers/kinetis/CMakeLists.txt]: I see, the order is by the config, not by driver souce code. Thanks for this check. updated.

**Comment by hakehuang** [comment]
> We need to have a corresponding PR in the main zephyr repo before we merge this
I see, I create pit driver pull request for zephyr below
https://github.com/zephyrproject-rtos/zephyr/pull/26629

File: mcux/drivers/imx/CMakeLists.txt [hunk 2]
```diff
@@ -15,6 +15,7 @@ zephyr_library_compile_definitions_ifdef(
 )
 
 zephyr_library_sources_ifdef(CONFIG_COUNTER_MCUX_GPT	fsl_gpt.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER_MCUX_PIT    fsl_pit.c)
```
Comment (MaureenHelm) [inline mcux/drivers/imx/CMakeLists.txt:18]: Sorry for the nitpick, but please replace the spaces with a tab to match the rest of the file.

File: mcux/drivers/kinetis/CMakeLists.txt [hunk 3]
```diff
@@ -15,6 +15,7 @@ zephyr_library_sources_ifdef(CONFIG_ADC_MCUX_ADC12	fsl_adc12.c)
 zephyr_library_sources_ifdef(CONFIG_ADC_MCUX_ADC16	fsl_adc16.c)
 zephyr_library_sources_ifdef(CONFIG_CAN_MCUX_FLEXCAN	fsl_flexcan.c)
 zephyr_library_sources_ifdef(CONFIG_COUNTER_MCUX_LPTMR	fsl_lptmr.c)
+zephyr_library_sources_ifdef(CONFIG_COUNTER_MCUX_PIT    fsl_pit.c)
```
Comment (MaureenHelm) [inline mcux/drivers/kinetis/CMakeLists.txt:18]: Sorry for the nitpick, but please replace the spaces with a tab to match the rest of the file.

File: mcux/drivers/imx/CMakeLists.txt (hunk 2 above)
Comment (hakehuang) [reply mcux/drivers/imx/CMakeLists.txt]: nice catch, my default setting is 4 whitespaces, this is not good for C code

File: mcux/drivers/kinetis/CMakeLists.txt (hunk 3 above)
Comment (hakehuang) [reply mcux/drivers/kinetis/CMakeLists.txt]: OK

**Pull request merged**